WalletWasabi.Fluent/ViewModels/Login/LoginViewModel.cs

```diff
@@ -49,11 +51,18 @@ public LoginViewModel(WalletViewModelBase walletViewModelBase)
 
 				if (!IsPasswordIncorrect)
 				{
-					wallet.Login();
-					walletViewModelBase.RaisePropertyChanged(nameof(WalletViewModelBase.IsLoggedIn));
 
-					// TODO: navigate to the wallet welcome page
-					Navigate().To(walletViewModelBase, NavigationMode.Clear);
+					var legalChecker = (await NavigationManager.MaterialiseViewModel(LegalDocumentsViewModel.MetaData) as LegalDocumentsViewModel)?.LegalChecker;
```

Comment:
NACK, this is a bad practice.

Reply:
there is no solution to this.

Reply:
only slightly tidier code.. but no solution.

Reply:
@molnard can we make LegalChecker accessible as a property on the wallet.. like Synchroniser or a property of Synchroniser or something?

Reply:
ok iv added a way for LegalChecker to be accessed from Synchronizer property on Wallet.

Significant refactoring is also needed and removed a load of buisness logic from legal docs viewmodels

Will do a pair programming with @soosr tomorrow to lick this into shape.

Reply:
We don't want to have Synchronizer as the new Global. I see two solutions here: 

- Pass down Legalchecker in the constructors.
- Start to use a dependency injection container. 
- Refactor the logic of view models. There might be a conceptual problem there - passing a business logic object should not be so hard to do.

Reply:
ok we found a much better solution.